services/report/raw_upload_processor.py

```diff
@@ -164,6 +185,24 @@ def _adjust_sessions(original_report, to_merge_report, to_merge_session, current
         for f in flags_under_carryforward_rules
         if f not in to_partially_overwrite_flags
     ]
+    if upload is None and to_partially_overwrite_flags:
+        log.warning("Upload is None, but there are partial_overwrite_flags present")
+    if (
+        upload
+        and USE_LABEL_INDEX_IN_REPORT_PROCESSING_BY_REPO_SLUG.check_value(
+            repo_slug(upload.report.commit.repository), default=False
+        )
+        and to_partially_overwrite_flags
+    ):
+        label_index_service = LabelsIndexService(upload.report)
+        # TODO: Needs shared upload
+        # if original_report._labels_index is None:
```

Comment:
If `_labels_index` is meant to be accessed from the outside then maybe remove the leading `_` (it's convention for private methods and attrs)

Reply:
That's fair... I'll change that in `shared` and adjust in the PR that actually makes use of it